### ISSUE TITLE
When Will the New Resource xClusterAzureVM will be live?

### ISSUE THREAD

**Issue by Anmolgan81** [PowerShell/xFailOverCluster#173]
For quite sometime we were dealing with a issue with the failovercluster, As we were not able to add second node to the cluster using the DSC configurations, We then Launch a support request for this, and for quite sometime lets say Oct-Dec, we were debugging the issue with the DSC PG team and the Cluster PG team, At the end the DSC PG team remade a template and we tested that and we were able to join the nodes to the cluster but the IP which the cluster was using was the Link Local IP, hence shared shared routatable IP's was not available for azure virtual machines.

The microsoft support has given us the below resolution for this:-
""As a resolution, the cluster team in Microsoft has recommended that we attempt the scenario using a link-local IPv6 shared IP and the DSC PowerShell team will submit a pull request to the community maintained xFailoverClustering module to add a new resource/module “xClusterAzureVM”. 
 
**There is no SLA for this contribution because it is community maintained and we will update them but its upto them when it gets published.
You can reach out to the maintainers directly over GitHub.**
 
We will submit a new resource to the open source maintainers for xFailoverCluster (something like xClusterAzureVM) using the code from the quickstart template.""

Is anybody know when the template will be out so that we can test it right away?

**Comment by johlju** [comment]
I got a hint that something was going on around this last DSC Community Call. @mgreenegit talked about this today at the DSC Community Call; you can listen back at the recording once it published. The gist of it; @mgreenegit will pull the resource out of the Azure template, refactor it a slightly (IPv4 -> IPv6), then it's going to be evaluated by the cluster team, and then sent in here for review. 

I imagine this will take weeks, exactly how long can't be determined today, but it will get here.

**Comment by johlju** [comment]
@mgreenegit we can use this issue to track this new resource xClusterAzureVM (or whatever the final name will be).

**Comment by Anmolgan81** [comment]
@johlju How canI listen to the skype meeting??

**Comment by johlju** [comment]
@Anmolgan81 It should be published here eventually [YouTube 
PowerShell and DSC Team Channel](https://www.youtube.com/channel/UCMhQH-yJlr4_XHkwNunfMog)

**Comment by Anmolgan81** [comment]
@johlju The recording is not available there?
Is there any timeline when the resource will get published??

**Comment by johlju** [comment]
No, the recording is not published there yet. I’m waiting for it myself. 

There is no timeline since most of this work is done on a voluntary basis. We all work together to improve these resources. 
Michael talked about getting that resource from the Azure template over here, so that will happen, but we should not expect that there will be a specific timeline for that.

**Comment by johlju** [comment]
@Anmolgan81 The recording is up now.

**Comment by Anmolgan81** [comment]
@johlju I'll check it thanks for letting me know.

**Comment by Anmolgan81** [comment]
I have listen to the recording.

So were are we with the new resource, is it coming this week? 
@johlju Can you tell me who will be best person to speak about the new resource so that I can know the status of the new resource ?

**Comment by johlju** [comment]
@Anmolgan81 this is work that @mgreenegit was looking into, to help move the resource here, and he is probably doing this on his free time as most of us do in the resource modules. Since he have not submitted something yet we should not expect anything.

If you have a need for this resource maybe you could help by look into making the resource yourself and send in a pull request (PR) here? I think the base he was gonna use is here and is under MIT license; https://github.com/Azure/azure-quickstart-templates/blob/master/sqlvm-alwayson-cluster/scripts/CreateFailoverCluster.ps1.zip

**Comment by Anmolgan81** [comment]
@johlju I dont have much knowledge on DSC but Ill try to provide whatever I can here.

**Comment by Anmolgan81** [comment]
@johlju @mgreenegit 
Hi Everyone,

So where are we with the modules??
are those are available for the community?
Where can I check the modules if they are available or not?

**Comment by mgreenegit** [comment]
@Anmolgan81, see the link to the zip file 4 posts above this one.  If all goes well I might be able to help out this weekend or early next week.  This isn't going to be a ton complex work, it just takes time.

**Comment by Anmolgan81** [comment]
Hi @mgreenegit, I have checked the link so we are going to build the new modules with the link that @johlju has provided am I right here?

**Comment by Anmolgan81** [comment]
@johlju @mgreenegit Any updates on this, where are we with the new modules?

**Comment by johlju** [comment]
@Anmolgan81 what I understood the new resource would be based on the code in the link I provided. On the community call @mgreenegit said that this still was on the todo list. If someone in the community want to work on this, then this is up for grabs.

**Comment by mgreenegit** [comment]
To move this forward, how about I create a branch and bring the new resources in for test/review and start tracking issues?  Then we can drive forward in small increments.

**Comment by Anmolgan81** [comment]
Am up for it, whatever I can help to get this deployed as soon as possible.

**Comment by johlju** [comment]
@mgreenegit Sounds good to me.

**Comment by bdanse** [comment]
Since this issue is still open are there alternative solutions for this issue? 

If not could someone point me in the right direction with regards to the IPv6 link local address?
